### PR TITLE
feat(Memory): add 'facts' attribute and improve code quality

### PR DESCRIPTION
**Pull request by danielchalef**
> :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit a3daf4dafe35adeffb208df86dd259ce526574ca.

### Summary:
This PR introduces a new 'facts' attribute to the 'Memory' class, updates tests to validate this attribute, and improves code quality in terms of string formatting, error messages, and comments.

**Key points**:
- Added a new attribute `facts` to the `Memory` class in `/zep_python/memory/models.py`.
- Updated tests in `/tests/memory_test.py` to validate the new `facts` attribute.
- Improved string formatting, error messages, and comments in multiple files.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: zep_python/memory/client.py [hunk 1]
```diff
@@ -41,29 +39,6 @@ def __init__(self, aclient: httpx.AsyncClient, client: httpx.Client) -> None:
         self.aclient = aclient
         self.client = client
 
-    def _parse_get_memory_response(self, response_data: Any) -> Memory:
```
Comment (paul-paliychuk) [inline zep_python/memory/client.py:44]: why did you remove this method?

Comment (danielchalef) [reply zep_python/memory/client.py:44]: It didn't appear to be doing anything more than pydantic does when parsing data into a model. I can't recall why we needed it to begin with :-/

**Review by paul-paliychuk — APPROVED** [review]

File: tests/memory_test.py [hunk 2]
```diff
@@ -122,6 +130,17 @@ async def test_aget_memory_missing_values(httpx_mock: HTTPXMock):
...
 
+@pytest.mark.asyncio
+async def test_aget_memory_with_facts(httpx_mock: HTTPXMock):
+    session_id = str(uuid4())
+
+    async with ZepClient(**mock_auth) as client:
+        httpx_mock.add_response(status_code=200, json=mock_messages_with_facts)
+        memory = await client.memory.aget_memory(session_id)
+
+        validate_memory(memory, with_facts=True)
```
[the remark this answers is not in the document]
Comment (danielchalef) [reply tests/memory_test.py:141]: validate_memory validates it
